File: step32_elastic_file_storage/efs-with-lambda/cdk.out/asset.32bfdff99c6e358f7f5e923e80d64fa6d3d90d187a3c9ebb45f2d6eaa4a803e6/msg.ts
const fs = require("fs").promises;

const MSG_FILE_PATH = "/mnt/msg/content";

exports.handler = async (event: any) => {
  console.log(event);
  const method = event.requestContext.http.method;

  if (method === "GET") {
    return sendRes(200, await getMessages());
  } else if (method === "POST") {
    await createMessage(event.body);
    return sendRes(200, await getMessages());
  } else if (method === "DELETE") {
    await deleteMessages();
    return sendRes(200, await getMessages());
  } else {
    return sendRes(200, "Method unsupported");
  }
};

const createMessage = async (message: string) => {
  try {
    await fs.appendFile(MSG_FILE_PATH, message + "\n");
  } catch (error) {
    console.log("error in creating msg", error);
  }
};
const getMessages = async () => {
  try {
    return await fs.readFile(MSG_FILE_PATH, "utf8");
  } catch (error) {
    console.log("error in getting messages", error);
  }
};

const deleteMessages = async () => {
  console.log("delete all messages");
  try {
    await fs.unlink(MSG_FILE_PATH);
  } catch (error) {
    console.log("error in deleting", error);
  }
};

const sendRes = (status: number, body: any) => {
  var response = {
    statusCode: status,
    body: body,
  };
  return response;
};

File: step32_elastic_file_storage/efs-with-lambda/Python/lambda/msg.py
from __future__ import print_function
import logging
import os
import json

msg_file_path = '/mnt/msg/content'


def handler(event, context):
    # request = event['requestContext']
    # http = request['http']
    method = event['requestContext']['http']['method']
    if method == 'GET':
        return getMessages()
    elif method == 'POST':
        message = json.loads(event['body'])
        return createMessages(message)
    elif method == 'DELETE':
        return deleteMessages()
    else:
        return {'message': 'method not supported'}


def getMessages():
    try:
        file = open(msg_file_path, 'r')
        file_text = file.read()
        return {'File_Text': file_text}
    except:
        logging.error('unable to read')
        return {'message': 'unable to load information'}


def deleteMessages():
    try:
        os.remove(msg_file_path)
        return {'message': 'File Deleted'}
    except:
        logging.error('unable to delete')
        return {'message': 'unable to load information'}


def createMessages(message):
    try:
        file = open(msg_file_path, 'a')
        file.write(message)
        return {'appended_text': message}
    except:
        logging.error('unable to write to the file')
        return {'message': 'unable to load information'}
